diff --git a/thirdparty/mbedtls/include/mbedtls/mbedtls_config.h b/thirdparty/mbedtls/include/mbedtls/mbedtls_config.h
index b757234a1c..9f7f095eae 100644
--- a/thirdparty/mbedtls/include/mbedtls/mbedtls_config.h
+++ b/thirdparty/mbedtls/include/mbedtls/mbedtls_config.h
@@ -4198,13 +4198,11 @@
  * This is the default value of ::mbedtls_platform_dev_random, which
  * can be changed at run time.
  */
-#ifdef OLDER_UNIX_ENABLED
-// Non-blocking for older Linux is less probably to lock-up.
-// Especially compiling with `-fipa-pta` enabled.
+// Always use `urandom` for randomness.
+// The concerns raised by mbedTLS do not apply for Godot,
+// and as the former blocks and will cause issues on older
+// environments with aggressive optimisation levels.
 #define MBEDTLS_PLATFORM_DEV_RANDOM "/dev/urandom"
-#else
-#define MBEDTLS_PLATFORM_DEV_RANDOM "/dev/random"
-#endif
 
 /** \def MBEDTLS_CHECK_RETURN
  *
