diff --git a/thirdparty/basis_universal/encoder/basisu_comp.cpp b/thirdparty/basis_universal/encoder/basisu_comp.cpp
index 81813257cd..fd2494a872 100644
--- a/thirdparty/basis_universal/encoder/basisu_comp.cpp
+++ b/thirdparty/basis_universal/encoder/basisu_comp.cpp
@@ -829,7 +829,7 @@ namespace basisu
 		return cECSuccess;
 	}
 
-	bool basis_compressor::generate_mipmaps(const imagef& img, basisu::vector<imagef>& mips, bool has_alpha)
+	bool basis_compressor::generate_mipmaps(const imagef& img, basisu::vector<imagef>& mips_levels, bool has_alpha)
 	{
 		debug_printf("basis_compressor::generate_mipmaps\n");
 
@@ -850,7 +850,7 @@ namespace basisu
 			const uint32_t level_width = maximum<uint32_t>(1, img.get_width() >> level);
 			const uint32_t level_height = maximum<uint32_t>(1, img.get_height() >> level);
 
-			imagef& level_img = *enlarge_vector(mips, 1);
+			imagef& level_img = *enlarge_vector(mips_levels, 1);
 			level_img.resize(level_width, level_height);
 
 			const imagef* pSource_image = &img;
@@ -858,7 +858,7 @@ namespace basisu
 			if (m_params.m_mip_fast)
 			{
 				if (level > 1)
-					pSource_image = &mips[level - 1];
+					pSource_image = &mips_levels[level - 1];
 			}
 
 			bool status = image_resample(*pSource_image, level_img, 
@@ -880,7 +880,7 @@ namespace basisu
 		return true;
 	}
 
-	bool basis_compressor::generate_mipmaps(const image &img, basisu::vector<image> &mips, bool has_alpha)
+	bool basis_compressor::generate_mipmaps(const image &img, basisu::vector<image> &mips_levels, bool has_alpha)
 	{
 		debug_printf("basis_compressor::generate_mipmaps\n");
 
@@ -915,7 +915,7 @@ namespace basisu
 			const uint32_t level_width = maximum<uint32_t>(1, img.get_width() >> level);
 			const uint32_t level_height = maximum<uint32_t>(1, img.get_height() >> level);
 
-			image &level_img = *enlarge_vector(mips, 1);
+			image &level_img = *enlarge_vector(mips_levels, 1);
 			level_img.resize(level_width, level_height);
 						
 			int result = stbir_resize_uint8_generic( 
@@ -940,7 +940,7 @@ namespace basisu
 			const uint32_t level_width = maximum<uint32_t>(1, img.get_width() >> level);
 			const uint32_t level_height = maximum<uint32_t>(1, img.get_height() >> level);
 
-			image& level_img = *enlarge_vector(mips, 1);
+			image& level_img = *enlarge_vector(mips_levels, 1);
 			level_img.resize(level_width, level_height);
 
 			const image* pSource_image = &img;
@@ -948,7 +948,7 @@ namespace basisu
 			if (m_params.m_mip_fast)
 			{
 				if (level > 1)
-					pSource_image = &mips[level - 1];
+					pSource_image = &mips_levels[level - 1];
 			}
 
 			bool status = image_resample(*pSource_image, level_img, m_params.m_mip_srgb, m_params.m_mip_filter.c_str(), m_params.m_mip_scale, m_params.m_mip_wrapping, 0, has_alpha ? 4 : 3);
diff --git a/thirdparty/basis_universal/encoder/basisu_comp.h b/thirdparty/basis_universal/encoder/basisu_comp.h
index 1cc75fc8a3..e018ff9f42 100644
--- a/thirdparty/basis_universal/encoder/basisu_comp.h
+++ b/thirdparty/basis_universal/encoder/basisu_comp.h
@@ -590,8 +590,8 @@ namespace basisu
 		bool write_output_files_and_compute_stats();
 		error_code encode_slices_to_uastc_hdr();
 		error_code encode_slices_to_uastc();
-		bool generate_mipmaps(const imagef& img, basisu::vector<imagef>& mips, bool has_alpha);
-		bool generate_mipmaps(const image &img, basisu::vector<image> &mips, bool has_alpha);
+		bool generate_mipmaps(const imagef& img, basisu::vector<imagef>& mips_levels, bool has_alpha);
+		bool generate_mipmaps(const image &img, basisu::vector<image> &mips_levels, bool has_alpha);
 		bool validate_texture_type_constraints();
 		bool validate_ktx2_constraints();
 		void get_dfd(uint8_vec& dfd, const basist::ktx2_header& hdr);
