diff --git a/thirdparty/astcenc/astcenc_weight_align.cpp b/thirdparty/astcenc/astcenc_weight_align.cpp
index b20541644a..683657dd79 100644
--- a/thirdparty/astcenc/astcenc_weight_align.cpp
+++ b/thirdparty/astcenc/astcenc_weight_align.cpp
@@ -209,22 +209,32 @@ static void compute_lowest_and_highest_weight(
 			vfloat svalrte = round(sval);
 			vfloat diff = sval - svalrte;
 			errval += diff * diff;
+			vfloat diff_two = vfloat(2.0f) * diff;
 
 			// Accumulate errors for minimum index
 			vmask mask = svalrte == minidx;
-			vfloat accum = cut_low_weight_err + vfloat(1.0f) - vfloat(2.0f) * diff;
+			vfloat accum = cut_low_weight_err + vfloat(1.0f) - diff_two;
 			cut_low_weight_err = select(cut_low_weight_err, accum, mask);
 
 			// Accumulate errors for maximum index
 			mask = svalrte == maxidx;
-			accum = cut_high_weight_err + vfloat(1.0f) + vfloat(2.0f) * diff;
+			accum = cut_high_weight_err + vfloat(1.0f) + diff_two;
 			cut_high_weight_err = select(cut_high_weight_err, accum, mask);
 		}
 
 		// Write out min weight and weight span; clamp span to a usable range
-		vint span = float_to_int(maxidx - minidx + vfloat(1));
-		span = min(span, vint(max_quant_steps + 3));
-		span = max(span, vint(2));
+		// Deconstruct SIMD constructors
+		vfloat span_f = maxidx - minidx;
+		span_f = span_f + vfloat(1.0f);
+		vint span = float_to_int(span_f);
+		
+		int max_quant_bound = max_quant_steps + 3;
+		vint v_max_quant = vint(max_quant_bound);
+		span = min(span, v_max_quant);
+		
+		vint v_min_quant = vint(2);
+		span = max(span, v_min_quant);
+		
 		storea(minidx, lowest_weight + sp);
 		storea(span, weight_span + sp);
 
